utils/converters/fbx/convert_to_threejs.py

```diff
@@ -1821,263 +1755,146 @@ def generate_mesh_object(node):
     else:
         attribute_type = (node.GetNodeAttribute().GetAttributeType())
         if attribute_type == FbxNodeAttribute.eMesh:
-            object_data = generate_mesh_object(node)
+            object_data = generate_mesh_object(node, geometry_dict, material_dict)
         elif attribute_type == FbxNodeAttribute.eLight:
             object_data = generate_light_object(node)
         elif attribute_type == FbxNodeAttribute.eCamera:
             object_data = generate_camera_object(node)
         else:
-            object_data = generate_object(node)
+            object_data = generate_generic_object(node)
 
-    object_count += 1
-    object_name = getObjectName(node)
+    object_data[uuid_key] = str(uuid.uuid4())
+    object_data[name_key] = get_object_name(node)
 
-    object_children = {}
-    for i in range(node.GetChildCount()):
-        object_count += generate_object_hierarchy(node.GetChild(i), object_children)
+    children = []
 
-    if node.GetChildCount() > 0:
-        # Having 'children' above other attributes is hard to read.
-        # We can send it to the bottom using the last letter of the alphabet 'z'. 
-        # This letter is removed from the final output.
-        if option_pretty_print:
-            object_data['zchildren'] = object_children
-        else:
-            object_data['children'] = object_children
+    for i in range(node.GetChildCount()):
+        children.append(generate_object(node.GetChild(i), geometry_dict, material_dict))
 
-    object_dict[object_name] = object_data
+    if len(children) > 0:
+        object_data[children_key] = children
 
-    return object_count
+    return object_data
 
-def generate_scene_objects(scene):
-    object_count = 0
-    object_dict = {}
+def generate_scene_object(scene, geometry_dict, material_dict):
+    children = []
 
     ambient_light = generate_ambient_light(scene)
     if ambient_light:
-        object_dict['AmbientLight'] = ambient_light
-        object_count += 1
+        children.append(ambient_light)
 
     if option_default_light:
-        default_light = generate_default_light()
-        object_dict['DefaultLight'] = default_light
-        object_count += 1
+        children.append(generate_default_light())
 
     if option_default_camera:
-        default_camera = generate_default_camera()
-        object_dict['DefaultCamera'] = default_camera
-        object_count += 1
+        children.append(generate_default_camera())
 
     node = scene.GetRootNode()
     if node:
         for i in range(node.GetChildCount()):
-            object_count += generate_object_hierarchy(node.GetChild(i), object_dict)
-
-    return object_dict, object_count
+            children.append(generate_object(node.GetChild(i), geometry_dict, material_dict))
+
+    return {
+      type_key: 'Scene',
+      uuid_key: str(uuid.uuid4()),
+      name_key: 'Scene',
+      'position': serialize_vector3((0,0,0)),
+      'rotation': serialize_vector3((0,0,0)),
```

Comment:
Fair call. I didn't really change as much stuff as it looks like I did :wink:

It was already using rotation, I just moved it to the new location:
https://github.com/mrdoob/three.js/pull/7046/files#diff-ccc8c41ed04893276377da090bf6f968L1980